packages/front-end/components/Segments/FactSegmentForm.tsx

```diff
@@ -0,0 +1,210 @@
+  );
+
+  // Filter the list of datasources to only show those that have atleast 1 fact built on it
+  const datasourceOptions = filteredDatasources.filter((filteredDs) =>
+    uniqueDatasourcesWithFactTables.includes(filteredDs.id)
+  );
+
+  const form = useForm({
+    defaultValues: {
+      name: current?.name || "",
+      datasource:
+        (current?.id ? current?.datasource : datasourceOptions[0]?.id) || "",
+      userIdType: current?.userIdType || "user_id",
+      owner: current?.owner || "",
+      description: current?.description || "",
+      factTableId: current?.factTableId || "",
+      filters: current?.filters || [],
+      type: "FACT",
+    },
+  });
+
+  const datasource = getDatasourceById(form.watch("datasource"));
+  const factTable = getFactTableById(form.watch("factTableId"));
+
+  return (
+    <Modal
+      close={close}
+      open={true}
+      size={"lg"}
+      cta={current?.factTableId ? "Update Segment" : "Create Segment"}
+      header={current?.factTableId ? "Edit Segment" : "Create Segment"}
+      submit={form.handleSubmit(async (value) => {
+        if (current?.id) {
+          await apiCall(`/segments/${current.id}`, {
+            method: "PUT",
+            body: JSON.stringify(value),
+          });
+        } else {
+          await apiCall(`/segments`, {
+            method: "POST",
+            body: JSON.stringify(value),
+          });
+        }
+        mutateDefinitions();
+      })}
+    >
+      <>
+        {!current?.id ? (
+          <div className="mb-3">
+            <a
+              href="#"
+              onClick={(e) => {
+                e.preventDefault();
+                goBack();
+              }}
+            >
+              <GBArrowLeft /> Go Back
+            </a>
+          </div>
+        ) : null}
+        <Field label="Name" required {...form.register("name")} />
+        <Field
+          label="Owner"
+          options={memberUsernameOptions}
+          comboBox
+          {...form.register("owner")}
+        />
+        <Field label="Description" {...form.register("description")} textarea />
+        <SelectField
+          label="Data Source"
+          required
+          value={form.watch("datasource")}
+          onChange={(v) => form.setValue("datasource", v)}
+          placeholder="Choose one..."
+          options={datasourceOptions.map((d) => ({
+            value: d.id,
+            label: `${d.name}${d.description ? ` — ${d.description}` : ""}`,
+          }))}
+          className="portal-overflow-ellipsis"
+          helpText="This list has been filtered to only show data sources that have at least one Fact Table built on top of it"
+        />
+        <SelectField
+          label="Identifier"
+          required
+          value={form.watch("userIdType")}
+          onChange={(v) => form.setValue("userIdType", v)}
+          placeholder="Select an identifier"
+          options={
+            datasource?.settings.userIdTypes?.map((userIdType) => ({
```

Comment:
These options should actually come from the selected fact table, not the data source. It's possible a fact table doesn't select all userIdType columns.

As such, probably want to put the fact table select above this